screens/Map.js

```diff
@@ -1,81 +1,82 @@
+    }
+}
+
 
 /**
  * US1 - As a user, I would like to navigate through SGW campus.
  * US2 - As a user, I would like to navigate through Loyola campus.
  * 
  * This is our main screen which includes all the components inside a map.
- * FIXME: When the app refreshes, a random component appears on in the upper
- *        half of the screen for miliseconds...
  */
-function Map ({ navigation }) {
-    const [switchVal, setswitchVal] = React.useState("");
-    const [getInsideBuild, setGetInsideBuild] = React.useState("");
-    const [mapPressed, setmapPressed] = React.useState("");
-
-    //TODO: To have a functionality for when the user presses on the map
-    AsyncStorage.setItem("mapPressed", mapPressed);
-
-    const campusSelected = async () => {
-        let tog = await AsyncStorage.getItem("toggle");
-        let inside = await AsyncStorage.getItem("getInsideBuilding");
-        setswitchVal(tog);
-        setGetInsideBuild(inside);
-    };
-
-    useEffect(() => {
-        const intervalId = setInterval(() => {
-            campusSelected();
-        }, 1);
-        return () => clearInterval(intervalId);
-    });
+export function Map (props) { 
 
+    const [isMapClicked, setIsMapClicked] = React.useState(false);
+    const mapRef = React.useRef(null);
+    
     return (
         <View testID="mapView" data-test="MapComponent">
             <View>
                 <MapView
                     data-test="MapViewComponent"
+                    ref={mapRef}
                     style={styles.map}
                     provider={PROVIDER_GOOGLE}
-                    region={switchVal === "true" ? mapPosition.sgwCoord : mapPosition.loyCoord}
+                    initialRegion={CampusRegion.sgwCoord}
                     showsUserLocation={true}
                     showsCompass={true}
                     showsBuildings={true}
                     showsIndoors={false}
+                    onPress = {() => { 
+                        if(isMapClicked) {
+                            setIsMapClicked(false);
+                            Keyboard.dismiss();
+                        }
+                        else {
+                            setIsMapClicked(true);
+                            Keyboard.dismiss();
+                            props.setSelectedBuildingName(null);
+                        }
+                    }}
+
                 >
                     <BuildingHighlight />
                     <BuildingIdentification />
+                    <LocationMarker mapReference = {mapRef} />
                 </MapView>
-                <Search testID="searchBar" navigation={navigation} />
+                <Search testID="searchBar" 
+                navigation = {props.navigation} 
+                mapReference = {mapRef} 
+                />
                 <View style={styles.CurrentBuildingLocation}>
-                        <CurrentBuildingLocation />
+                        {/* <CurrentBuildingLocation /> */}
```

Comment:
The current building location button should be uncommented.

Reply:
It creates warning that affects the performance of the app... Those warnings should be fix before

Reply:
What kind of warnings ? @AlainJobs